src/v/crypto/test/digest_tests.cc

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "crypto/crypto.h"
+#include "crypto/types.h"
+#include "crypto_test_utils.h"
+#include "gtest/gtest.h"
+#include "test_utils/test.h"
+
+#include <absl/container/flat_hash_map.h>
+#include <boost/algorithm/hex.hpp>
+#include <gtest/gtest.h>
+
+#include <iterator>
+#include <type_traits>
+
+const auto sha256_test_val = []() {
+    return convert_from_hex(
+      "09fc1accc230a205e4a208e64a8f204291f581a12756392da4b8c0cf5ef02b95");
+}();
```

Comment:
Does this immediately invoked function expression (IIFE) perform some trick I'm not aware of, or is it leftover from something more complex?

Reply:
Just holds the hex to byte conversion result for easy use in tests

Reply:
But this is the same, right:
```
const auto sha256_test_val = convert_from_hex(
  "09fc1accc230a205e4a208e64a8f204291f581a12756392da4b8c0cf5ef02b95");
```

Reply:
... yeah... sorry not sure what i was thinking